client/prioritizing.go

```diff
@@ -54,12 +54,10 @@ func (p *prioritizingClient) learnGroup(ctx context.Context) error {
 	var err error
 
 	for _, c := range p.clients {
-		if hc, ok := c.(*httpClient); ok {
-			chainInfo, err = hc.FetchChainInfo(p.chainHash)
-			if err == nil {
-				p.chainInfo = chainInfo
-				return nil
-			}
+		chainInfo, err = c.Info(ctx)
+		if err == nil {
+			p.chainInfo = chainInfo
```

Comment:
This is not protected by a lock, yet it is called from top-level methods like Watch and Info.
We should either document the client interface and say it is not re-entrant, or protect this.

Reply:
added lock